solidity/test/bridge/Bridge.Wallets.test.ts

```diff
@@ -315,22 +321,83 @@ describe("Bridge - Wallets", () => {
+                  await expect(
+                    bridge.requestNewWallet(NO_MAIN_UTXO)
+                  ).to.be.revertedWith("Wallet creation conditions are not met")
+                })
+              }
+            )
 
-            after(async () => {
-              await restoreSnapshot()
-            })
+            context(
+              "when the minimum BTC balance threshold is non-zero",
+              () => {
+                before(async () => {
+                  await createSnapshot()
+                })
+
+                after(async () => {
+                  await restoreSnapshot()
+                })
+
+                it("should revert", async () => {
+                  await expect(
+                    bridge.requestNewWallet(NO_MAIN_UTXO)
+                  ).to.be.revertedWith("Wallet creation conditions are not met")
+                })
+              }
+            )
 
-            // If the wallet has 0 BTC, the function must revert.
-            // This is because the active wallet's balance must be above the
-            // minimum BTC balance threshold and that threshold is guaranteed
-            // to be always greater than zero.
-            it("should revert", async () => {
-              await expect(
-                bridge.requestNewWallet(NO_MAIN_UTXO)
-              ).to.be.revertedWith("Wallet creation conditions are not met")
+            context("when the minimum BTC balance threshold is zero", () => {
+              context("when wallet creation conditions are met", () => {
+                let tx: ContractTransaction
+
+                before(async () => {
+                  await createSnapshot()
+
+                  // Set the minimum BTC balance to zero.
+                  await bridgeGovernance
+                    .connect(governance)
+                    .beginWalletCreationMinBtcBalanceUpdate(0)
+                  await helpers.time.increaseTime(constants.governanceDelay)
+                  await bridgeGovernance
+                    .connect(governance)
+                    .finalizeWalletCreationMinBtcBalanceUpdate()
+
+                  // Make the wallet old enough. Be absolutely sure - it is
+                  // possible that after the governance delay passed, the wallet
+                  // is already old enough.
+                  await increaseTime(constants.walletCreationPeriod)
+
+                  tx = await bridge.requestNewWallet(NO_MAIN_UTXO)
+                })
+
+                after(async () => {
+                  walletRegistry.requestNewWallet.reset()
+
+                  await restoreSnapshot()
+                })
+
+                it("should emit NewWalletRequested event", async () => {
+                  await expect(tx).to.emit(bridge, "NewWalletRequested")
+                })
+
+                it("should call ECDSA Wallet Registry's requestNewWallet function", async () => {
+                  await expect(walletRegistry.requestNewWallet).to.have.been
```

Comment:
```suggestion
                 expect(walletRegistry.requestNewWallet).to.have.been
```

Reply:
Done in fa2b374ef9e5abbbe9730262f1d550f14886b8ca.